binja_script.py

```diff
@@ -0,0 +1,400 @@
+        """
+        Sets the 'current pc' tag to the specified address, and clears the old ones.
+        """
+        self.clear_pc_tag()
+        add_auto_tag(self.bv, new_pc, "pwndbg-pc", "current pc")
+
+    @should_register
+    def get_bp_tags(self) -> List[int]:
+        """
+        Gets a list of all addresses with a breakpoint tag.
+        """
+        return [t.addr for t in get_tag_refs(self.bv, "pwndbg-bp")]
+
+    @should_register
+    def get_symbol(self, addr: int) -> str | None:
+        """
+        Gets the symbol at exactly the specified address.
+        """
+        sym = self.bv.get_symbol_at(addr)
+        if sym is None:
+            return None
+        return sym.full_name
+
+    @should_register
+    def get_func_info(self, addr: int) -> Tuple[str, int] | None:
+        """
+        Gets the widest function containing the specified address.
+
+        Returns a (function name, offset from start) tuple.
+        """
+        func = get_widest_func(self.bv, addr)
+        if func is None:
+            return None
+        return (func.symbol.full_name, func.start)
+
+    @should_register
+    def get_data_info(self, addr: int) -> Tuple[str, int] | None:
+        """
+        Gets the data variable containing the specified address.
+
+        Returns a (variable name, offset from start) tuple.
+        """
+        dv = self.bv.get_data_var_at(addr)
+        if dv is None:
+            return None
+        if dv.symbol is not None:
+            return (dv.symbol.full_name, dv.address)
+        return (dv.name or f"data_{dv.address:x}", dv.address)
+
+    @should_register
+    def get_comments(self, addr: int) -> List[str]:
+        """
+        Gets a list of all comments at a specified address.
+        """
+        ret = []
+        for f in sorted(self.bv.get_functions_containing(addr), key=lambda f: f.start):
+            ret += f.get_comment_at(addr).split("\n")
+        ret += self.bv.get_comment_at(addr).split("\n")
+        # remove empty lines and prepend double slash
+        return ["// " + x for x in ret if x]
+
+    @should_register
+    def decompile_func(
+        self, addr: int, level: str
+    ) -> List[Tuple[int, List[Tuple[str, str]]]] | None:
+        """
+        Gets the decompilation of a function at a specified IL level.
+
+        Returns a list of (address, token) tuples, where each token is a (text, type) tuple.
+        """
+        func = get_widest_func(self.bv, addr)
+        if func is None:
+            return None
+        orig_func = func
+        if level == "disasm":
+            pass
+        elif level == "llil":
+            func = func.llil_if_available
+        elif level == "mlil":
+            func = func.mlil_if_available
+        elif level == "hlil":
+            func = func.hlil_if_available
+        else:
+            raise ValueError(f"Cannot decompile with level {level!r}")
+        if func is None:
+            return None
```

Comment:
If the given IL is not available, does this mean we cannot decompile with that level and we should maybe raise a ValueError as well?

Reply:
I'm not sure the conditions where IL won't be available, but I don't think it warrants showing an error the user every single time they try to decompile anything.